src/utils/daemon.js

```diff
@@ -1,63 +1,104 @@
 import IPFSFactory from 'ipfsd-ctl'
 import logger from './logger'
+import { showRepoApiFileErrorMessage } from './errors'
+import { join } from 'path'
+import fs from 'fs-extra'
+import { spawnSync } from 'child_process'
+import findExecutable from 'ipfsd-ctl/src/utils/find-ipfs-executable'
 
-export default async function createDaemon (opts) {
-  opts.type = opts.type || 'go'
-  opts.path = opts.path || ''
-  opts.flags = opts.flags || ['--migrate=true', '--routing=dhtclient']
-  opts.keysize = opts.keysize || 4096
+function repoFsck (path) {
+  const exec = findExecutable('go', join(__dirname, '..'))
+  spawnSync(exec, ['repo', 'fsck'], {
+    env: {
+      ...process.env,
+      IPFS_PATH: path
+    }
+  })
+}
 
-  if (opts.type !== 'go') {
-    throw new Error(`${opts.type} connection is not supported yet`)
+async function configure (ipfsd) {
+  const cfgFile = join(ipfsd.repoPath, 'config')
+  const cfg = await fs.readJSON(cfgFile)
+
+  let origins = []
+  try {
+    origins = cfg.API.HTTPHeaders['Access-Control-Allow-Origin']
+  } catch (e) {
+    logger.warn(e)
   }
 
-  const factory = IPFSFactory.create({ type: opts.type })
+  if (!Array.isArray(origins)) {
+    origins = []
+  }
+
+  if (!origins.includes('webui://-')) origins.push('webui://-')
+  if (!origins.includes('https://webui.ipfs.io')) origins.push('https://webui.ipfs.io')
+
+  cfg.API.HTTPHeaders['Access-Control-Allow-Origin'] = origins
+  cfg.API.HTTPHeaders['Access-Control-Allow-Methods'] = ['PUT', 'GET', 'POST']
+
+  await fs.writeJSON(cfgFile, cfg)
+}
+
+async function spawn ({ type, path, keysize = 0 }) {
```

Comment:
What happens if `keysize` is set to 0, and why set a default for it here?

Reply:
`keysize` is only used when initializing a repo. We don't actually need to set it to 0 here. The default is set on store.js when we add the config if it's not present.